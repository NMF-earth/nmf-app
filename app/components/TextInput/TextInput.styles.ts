import { StyleSheet } from "react-native";
import { Colors } from "../../style";
import { PADDING_HORIZONTAL } from "../../constants/Layout";

export default StyleSheet.create({
  container: {
    marginVertical: 12,
    marginHorizontal: PADDING_HORIZONTAL
  },
  input: {
    paddingHorizontal: PADDING_HORIZONTAL,
    height: 40,
    borderColor: Colors.gray,
    borderWidth: 2,
    borderRadius: 20
  },
  containerOptionalTitle: {
    marginBottom: 14,
    flexDirection: "row"
  },
  icon: {
    marginRight: 10
  }
});
