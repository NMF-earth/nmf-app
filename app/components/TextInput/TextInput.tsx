import React, { useState } from "react";
import { TextInput as TextInputRN, View, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { Colors } from "style";

import Text from "../Text";
import styles from "./TextInput.styles";

interface Prop {
  isOptional: boolean;
  placeholder?: string;
  title: string;
  value: string;
  onChangeText: (text: string) => void;
}
const TextInput = ({
  isOptional,
  onChangeText,
  value,
  placeholder,
  title,
}: Prop) => {
  const [isVisible, setIsVisible] = useState(isOptional ? false : true);

  return (
    <View style={styles.container}>
      <TouchableOpacity
        activeOpacity={isOptional ? 0.2 : 1}
        onPress={() => (isOptional ? setIsVisible(!isVisible) : {})}
        style={styles.containerOptionalTitle}
      >
        {isOptional && (
          <Ionicons
            name={"md-add-circle"}
            size={24}
            color={Colors.green50}
            style={styles.icon}
          />
        )}
        <Text.H3 green={isOptional}>{title}</Text.H3>
      </TouchableOpacity>
      {isVisible && (
        <TextInputRN
          placeholder={placeholder}
          style={styles.input}
          onChangeText={(text) => onChangeText(text)}
          value={value}
        />
      )}
    </View>
  );
};

export default TextInput;
