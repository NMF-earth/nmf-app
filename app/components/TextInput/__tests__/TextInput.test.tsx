import React from "react";
import renderer from "react-test-renderer";
import TextInput from "../";

jest.unmock("../");

const foo = () => {
  // do nothing.
};

it("renders correctly TextInput", () => {
  const tree = renderer
    .create(
      <TextInput
        isOptional={false}
        placeholder={"placeholder"}
        title={"title"}
        onChangeText={foo}
        value={"value"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly TextInput if optional", () => {
  const tree = renderer
    .create(
      <TextInput
        isOptional={false}
        placeholder={"placeholder"}
        title={"title"}
        onChangeText={foo}
        value={"value"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
