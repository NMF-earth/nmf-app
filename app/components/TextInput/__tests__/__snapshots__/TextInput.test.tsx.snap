// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly TextInput 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 14,
      "marginVertical": 12,
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 14,
      }
    }
  >
    <Text.H3
      green={false}
    >
      title
    </Text.H3>
  </TouchableOpacity>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="placeholder"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#CED4DA",
        "borderRadius": 20,
        "borderWidth": 2,
        "height": 40,
        "paddingHorizontal": 14,
      }
    }
    underlineColorAndroid="transparent"
    value="value"
  />
</View>
`;

exports[`renders correctly TextInput if optional 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 14,
      "marginVertical": 12,
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 14,
      }
    }
  >
    <Text.H3
      green={false}
    >
      title
    </Text.H3>
  </TouchableOpacity>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="placeholder"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#CED4DA",
        "borderRadius": 20,
        "borderWidth": 2,
        "height": 40,
        "paddingHorizontal": 14,
      }
    }
    underlineColorAndroid="transparent"
    value="value"
  />
</View>
`;
